script.js

```diff
@@ -1 +1,129 @@
 //😍CEOS 20기 프론트엔드 파이팅😍
+
+document.addEventListener("DOMContentLoaded", function () {
+    const today = new Date(); // 현재 날짜와 시간을 가져오는 Date 객체
+    const options = { month: "long", day: "numeric", weekday: "long" };
+    // 날짜, 요일 등 포맷 시 month와 weekday는 긴 형식으로 (9월, 목요일) day는 숫자 형식 (5, 25)
+
+    const formattedDate = today.toLocaleDateString("ko-KR", options); // options 형식의 한국어 날짜
+    document.querySelector(".date").textContent = formattedDate;
+    // .Date 요소의 textcontent를 formattedDate으로 설정
+  
+    const todoInput = document.querySelector("input"); // input 요소 가지고 오기
+    const todoBox = document.querySelector(".todoBox"); // class가 .todo-box인 요소를 가지고 오기
+    const submitBtn = document.getElementById("submitBtn"); // id가 submitbtn인 요소를 가지고 오기
+    const completedCountElem = document.querySelector(".completedCount"); // 완료된 todo 수를 셀 요소
+    const totalCountElem = document.querySelector(".totalCount"); // 전체 todo 수를 셀 요소
+  
+    let todoList = []; // 로컬스토리지에 저장될 todo 배열
+
+    function setting() {
+      loadStorage(); // localStorage에 저장된 todoList의 todo들 불러오기
+      submitBtn.addEventListener("click", function (event) {
+          event.preventDefault(); // input에 값 입력 후 추가 버튼을 눌러도 새로고침 되지 않도록.
+          createList();
+      });
+  }
+
+  function createList() {
+      const newTodo = todoInput.value.trim(); // 문자열 앞 뒤 공백을 제거하는 trim을 이용, 사용자가 input에 입력한 todo를 저장
+      if (newTodo === ""){
+        alert('오늘의 할 일을 적어주세요!🍀');
+        return; // 사용자가 입력하지 않았으면 함수 종료 
+      }
+        // 이미 같은 내용의 투두가 있는지 확인
+
+        const isDuplicate = todoList.some((todo) => todo.text === newTodo); //some 메서드, 배열의 각 요소를 순회하면서, 주어진 조건을 만족하는 요소가 하나라도 있으면 true를 반환하고, 조건을 만족하는 요소가 없으면 false를 반환
+        if (isDuplicate) {
+            alert("이미 동일한 투두가 있습니다!👏🏻"); // 알림창으로 중복 투두 알림
+            todoInput.value = "";
+            return; // 중복되면 함수 종료
+        }
+
+      todoList.push({ text: newTodo, completed: false }); // 배열에 입력 값 저장 
```

Comment:
현재는 todoList의 항목이 텍스트 값만으로 관리되고 있는 것 같아요. 물론 그에 대한 중복 입력 예외처리가 되어있지만,  하지만 일부 상황에서는 같은 이름의 할 일을 여러 번 추가하고 싶을 수 있음을 고려하면 **각 할 일에 고유 ID를 추가**하는 방법을 사용하면 해결할 수 있어요! 텍스트 중복을 허용하면서도 할 일의 개별적인 상태(완료 여부 등)를 관리할 수 있을 것 같아요 🤩

```suggestion
todoList.push({ id: Date.now(), text: newTodo, completed: false });

```

Reply:
중복 입력 예외 처리를 한 이유가 고유 ID가 아닌 text로만 todo들을 관리하다 보니 중복 todo가 생겼을 때 하나의 todo를 완료 및 삭제를 해도 다른 중복 todo까지 동일한 이벤트가 적용되어 이 버그의 해결책으로 some()메서드를 이용해 같은 text를 가진 todo가 있으면 입력을 방지하는 것을 선택했었습니다 😂😂

근데 정말 중요한 todo라서 잊지 않기 위해 여러 개 중복 입력할 수도 있을 테고, 또 좀 더 복잡한 중복 입력 예외처리 보다는 고유 ID를 추가해 더 간결하게 기존에 발생했던 버그를 해결할 수 있을 것 같아 고유 ID 추가하는 방식으로 수정하겠습니다!
알려주셔서 감사합니다 🔥👍🏻👍🏻